addons/sale_mrp/tests/test_sale_mrp_flow.py

```diff
@@ -0,0 +1,334 @@
+        # -------------------------------------------
+        product_a = self.Product.create({
+            'name': 'Product A',
+            'type': 'product',
+            'uom_id': self.uom_unit.id,
+            'uom_po_id': self.uom_unit.id,
+            'route_ids': [(6, 0, [route_manufacture, route_mto])]})
+
+        # Bill of materials for Product A.
+        bom_a = self.MrpBom.create({
+            'name': product_d.name,
+            'product_tmpl_id': product_a.product_tmpl_id.id,
+            'product_qty': 2,
+            'product_uom': self.uom_dozen.id,
+            'product_efficiency': 1.0,
+            'type': 'normal'})
+
+        self.MrpBomLine.create({
+            'product_id': product_b.id,
+            'product_qty': 3,
+            'type': 'phantom',
+            'bom_id': bom_a.id,
+            'product_uom': self.uom_unit.id})
+
+        self.MrpBomLine.create({
+            'product_id': product_c.id,
+            'product_qty': 300.5,
+            'type': 'normal',
+            'bom_id': bom_a.id,
+            'product_uom': self.uom_gm.id})
+
+        self.MrpBomLine.create({
+            'product_id': product_d.id,
+            'product_qty': 4,
+            'type': 'phantom',
+            'bom_id': bom_a.id,
+            'product_uom': self.uom_unit.id})
+
+        self.MrpBomLine.create({
+            'product_id': product_d.id,
+            'product_qty': 4,
+            'type': 'normal',
+            'bom_id': bom_a.id,
+            'product_uom': self.uom_unit.id})
+        # ----------------------------------------
+        # Create Sale order of 10 Dozen product A.
+        # ----------------------------------------
+        order = self.SaleOrder.create({
+            'partner_id': self.partner_agrolite.id,
+            'date_order': datetime.today(),
+        })
+        self.SaleOrderLine.create({
+            'order_id': order.id,
+            'product_id': product_a.id,
+            'product_uom_qty': 10,
+            'product_uom': self.uom_dozen.id
+        })
+        assert order, "Sale order not created."
+        context = {"active_model": 'sale.order', "active_ids": [order.id], "active_id": order.id}
+        order.with_context(context).action_button_confirm()
+
+        # Run procurement.
+        # ---------------
+        proc_ids = self.ProcurementOrder.search([('origin', 'like', order.name)])
+        proc_ids.run()
+
+        # ----------------------------------------------------
+        # Check production order for product A.
+        # ----------------------------------------------------
+
+        mo_a = self.ProductionOrder.search([('origin', 'like', order.name), ('product_id', '=', product_a.id)])
+        self.assertEqual(len(mo_a), 1, 'Production order not created.')
+        # Check quantity, unit of measure and state of production order.
+        self.assertEqual(mo_a.product_id.id, product_a.id, 'Wrong product in production order.')
```

Comment:
You search on product first and then check if it is equal.   The test is a little unnecessary.